File: core/js/src/legacy_api.ts
export function patchLegacyRecord(r: Record<string, any>): Record<string, any> {
  if (!("dataset_id" in r)) {
    return r;
  }
  const record = { ...r };
  if (!("expected" in record)) {
    record.expected = record.output;
  }
  delete record.output;
  return record;
}

export function makeLegacyRecord(r: Record<string, any>): Record<string, any> {
  if (!("dataset_id" in r)) {
    return r;
  }
  const record = { ...r };
  if (!("output" in record)) {
    record.output = record.expected;
  }
  delete record.expected;
  return record;
}

File: core/py/src/braintrust_core/legacy_api.py
import json


def patch_legacy_dataset(r):
    row = {**r}
    output = row.pop("output", None)
    if "expected" not in row:
        row["expected"] = output
    return row


def patch_legacy_record(r):
    if "dataset_id" not in r:
        return r
    return patch_legacy_dataset(r)


def make_legacy_dataset(r):
    row = {**r}
    expected = row.pop("expected", None)
    if "output" not in row:
        row["output"] = expected
    return row


def make_legacy_record(r):
    if "dataset_id" not in r:
        return r
    return make_legacy_dataset(r)
